lib/navigation/preprocess_encoders.cpp

```diff
@@ -1,91 +1,94 @@
+#include "consts.hpp"
 #include "preprocess_encoders.hpp"
 
+#include "core/types.hpp"
+
 namespace hyped::navigation {
 
 EncodersPreprocessor::EncodersPreprocessor(core::ILogger &logger)
     : logger_(logger),
-      encoder_outliers_({0, 0, 0, 0}),
-      encoders_reliable_({true, true, true, true}),
-      num_reliable_encoders_(core::kNumImus)
+      num_consecutive_outliers_per_encoder_({0, 0, 0, 0}),
+      is_reliable_per_encoder_({true, true, true, true}),
+      num_reliable_encoders_(core::kNumImus),
+      max_consecutive_outliers_(10)
 {
 }
 
 std::optional<core::EncoderData> EncodersPreprocessor::processData(
-  const core::EncoderData raw_encoder_data)
+  const core::EncoderData &raw_encoder_data)
 {
-  const auto encoder_data = detectOutliers(raw_encoder_data);
-  SensorChecks sensors    = checkReliable();
-  if (!encoder_data.has_value() || sensors == SensorChecks::kUnacceptable) {
-    return std::nullopt;
-  } else {
-    return encoder_data;
-  }
+  const auto encoder_data = sanitise(raw_encoder_data);
+  const auto sensors      = checkReliable();
```

Comment:
Why `auto` if we know return type?

Reply:
we always know the return type, for every function call. auto merely leaves it to the compiler to fill in what is known.

i prefer in most cases because it allows us to change the return types without breaking lots of code (that is return-type agnostic)

the exceptions are 
1. primitive types such as `bool` and `core::Float`
2. if we are converting types, e.g. to an interface. think 
   ```cpp
    const sensors::ICan can = sensors::HardwareCan::create();
    ```